### PR TITLE
[performance] Speed up some of the slower db queries

### PR DESCRIPTION
**Pull request by tsmethurst**
This PR:

1. Removes use of the `LOWER()` db function where it wasn't necessary (this function prevented the db indexes from working properly).
2. Swaps the db exists function from using `COUNT(*)` to using `EXISTS`, which should be faster.
3. Reworks the blocked function to always present params in the order that's been indexed.
4. Adds a warning level log when db queries take longer than 1s to help debugging.

closes https://github.com/superseriousbusiness/gotosocial/issues/520

### CODE REVIEW

**Comment by igalic** [comment]
doesn't this still need migrations for usernames?

**Comment by tsmethurst** [comment]
Aha see the addition I made to the issue. I just realized that all the usernames we use for accounts on a GtS instance are lowercase, so it should be fine.

File: internal/db/bundb/status.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (s *statusDB) GetStatusByID(ctx context.Context, id string) (*gtsmodel.Stat
 			return s.cache.GetByID(id)
 		},
 		func(status *gtsmodel.Status) error {
-			return s.newStatusQ(status).Where("LOWER(status.id) = LOWER(?)", id).Scan(ctx)
+			return s.newStatusQ(status).Where("status.id = ?", id).Scan(ctx)
```
Comment (NyaaaWhatsUpDoc) [inline internal/db/bundb/status.go:73]: do you need to do a `strings.ToLower(id)` etc on all of these changes in this file for the query argument?

File: internal/db/bundb/status.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ func (s *statusDB) GetStatusByURI(ctx context.Context, uri string) (*gtsmodel.St
 			return s.cache.GetByURI(uri)
 		},
 		func(status *gtsmodel.Status) error {
-			return s.newStatusQ(status).Where("LOWER(status.uri) = LOWER(?)", uri).Scan(ctx)
+			return s.newStatusQ(status).Where("status.uri = ?", uri).Scan(ctx)
```
Comment (NyaaaWhatsUpDoc) [inline internal/db/bundb/status.go:85]: Where statuses are put into the database, we should probably force a `strings.ToLower(...)` on their URI/URL, and do the same for the incoming parameter coming in here (unless you've already included that somewhere else)

**Pull request merged**